examples/covariance/plot_outlier_detection.py

```diff
@@ -72,19 +79,23 @@
     X = np.r_[X, np.random.uniform(low=-6, high=6, size=(n_outliers, 2))]
 
     # Fit the model
-    plt.figure(figsize=(10.8, 3.6))
+    plt.figure(figsize=(9, 7))
     for i, (clf_name, clf) in enumerate(classifiers.items()):
         # fit the data and tag outliers
-        clf.fit(X)
-        scores_pred = clf.decision_function(X)
+        if clf_name == "Local Outlier Factor":
```

Comment:
Predicting on the **testing set** does not work for `KNeighbors`... That is why `_predict` is private (but `decision_function` isn't, since we need it in this exmple for instance)